### DISCUSSION TITLE
Flashing Status LED

### DISCUSSION THREAD

**Discussion by kevindelord** [MiczFlor/RPi-Jukebox-RFID#2242] [Q&A] [answered]
Hello there, 
I'm not sure if it's a bug or not quite yet, hence the discussion thread.

When I press the shutdown button the status LED does not flash ("on/off/on/off").
Isn't this supposed to be the case? Only on shutdown or also on startup? 

If it's possible... how to? 

Here is my gpio setup:
```
[Shutdown]
enabled: True
Type: Button
Pin: 3
hold_mode: Postpone
hold_time: 2.0
functionCall: functionCallShutdown

[StatusLED]
enabled: True
Type: StatusLED
Pin: 26
```

Thanks!

**Comment by Groovylein** [comment, marked as answer]
Hey,

Do you mean the OnOffShim? 
Did you install the software for the OnOffShim (see https://splittscheid.de/phoniebox-bauanleitung-toniebox-alternative/#ftoc-installation-onoffshim)?

**Reply by AlvinSchiller** [reply]
Also here ;)
https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3/main/documentation/builders/components/power/onoff-shim.md

**Reply by kevindelord** [reply]
I will look into that, thanks!

**Reply by AlvinSchiller** [reply]
This might be also interesting for you? https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/1983
Don't know if it hits your case.

**Reply by kevindelord** [reply]
Hello both of you :) 

I now have the status LED flashing when pressing the button to initiate shutdown, it then stays on during "clean" shutdown, flashes again and finally the system turns off. It's pretty good I think! 

On the links you sent above there is a script / code to make the LED flash during power-up. 
```
Installation Buttons LED Service
Mit folgenden Befehlen kann mein Script zum animierten Start der Phoniebox installiert werden. Dabei ist zu beachten, dass die Angabe der GPIOs in der gpiozero_led.py angepasst werden müssen!!!

git clone https://github.com/splitti/phoniebox_led_control --branch master
sudo ln -s ~/phoniebox_led_control/service/phoniebox_led_control.service /etc/systemd/system/phoniebox_led_control.service
sudo systemctl daemon-reload 
sudo systemctl enable /etc/systemd/system/phoniebox_led_control.service
sudo service phoniebox_led_control start
```

I looked into the code but I don't think it works for the for "status" LED. 
Have you been able to use this code successfully?

**Comment by kevindelord** [comment]
If anyone comes here again, here is how I've managed to makes the status LED flash on startup.
Note: the GPIO are not accessible until a certain point during boot, meaning it does not seem possible to make the LED flash right away but after few seconds. 

File: `/home/pi/phoniebox_led_control/led_startup.py`

```
#!/usr/bin/python3

import RPi.GPIO as GPIO
import time
import sys
import subprocess

LED_PIN = 26

GPIO.setmode(GPIO.BCM)
GPIO.setup(LED_PIN, GPIO.OUT)

def getshell():
    process = subprocess.Popen("echo -e status\\nclose | nc -w 1 localhost 6600 | grep 'OK MPD'", shell=True, stdout=subprocess.PIPE).communicate()[0].decode('utf-8').strip()
    return process

process = ""
while process == "":
    GPIO.output(LED_PIN, GPIO.LOW)
    time.sleep(0.3)
    GPIO.output(LED_PIN, GPIO.HIGH)
    time.sleep(0.3)
    process = getshell()
```

File: `/etc/systemd/system/phoniebox-led-control.service` 

```
[Unit]
Description=Phoniebox LED-Buttons-Control Service
DefaultDependencies=false

[Service]
WorkingDirectory=/home/pi/phoniebox_led_control
ExecStart=/usr/bin/python3 /home/pi/phoniebox_led_control/led_startup.py

[Install]
WantedBy=multi-user.target
```

And finally to activate the LED during start:
```
$> sudo systemctl daemon-reload 
$> sudo systemctl enable /etc/systemd/system/phoniebox-led-control.service
$> sudo service phoniebox-led-control start
```
